src/dmd/aliasthis.d

```diff
@@ -105,8 +127,1066 @@ extern (C++) Expression resolveAliasThis(Scope* sc, Expression e, bool gag = fal
             e = e.expressionSemantic(sc);
         }
         e = resolveProperties(sc, e);
-        if (gag && global.endGagging(olderrors))
-            e = null;
     }
     return e;
 }
+
+/*
+ * Similar resolveAliasThis, but it doesn't resolve properties and templates.
+ */
+private Expression resolveAliasThis2(Scope* sc, Expression e)
```

Comment:
I really don't like the `function2` naming convention used in DMD.  I know such names are used elsewhere in DMD.  Perhaps add a `bool` parameter to `resolveAliasThis`.  It's just a suggestion, though; I'd still merge this accept this even if no changes were made.

Reply:
fixed